### DISCUSSION TITLE
How to add pkcs12 certificates for API automation

### DISCUSSION THREAD

**Discussion by praveen222** [webdriverio/webdriverio#6813] [Community]
Hi  i am using request method for API automation. My API's are Authenticated via keystore certificates [.jks].

can some one guide me how to  pass certificates in webdriverio API automation

**Comment by christian-bromann** [comment]
@praveen222 I am not sure if I can follow. Are you trying to test your API which requires authentification via keystore certificates? If so I recommend to use just a normal unit test framework like Mocha or Jest as you don't need browser automation capabilities to do that.

**Reply by praveen222** [reply]
Hi @christian-bromann : you are correct - i am trying to test your API which requires authentication via keystore certificates

I need to know how to pass the keystore certificate in code base

**Reply by christian-bromann** [reply]
For pure API testing you don't need a browser. Therefor I recommend to use a test framework such as Mocha or Jest and a request library such as `got` as your setup.
